src/Features/Core/Portable/ExtractMethod/SelectionResult.cs

```diff
@@ -138,6 +139,61 @@ public bool ShouldPutAsyncModifier()
             return false;
         }
 
+        public bool ShouldCallConfigureAwaitFalse()
+        {
+            var firstToken = GetFirstTokenInSelection();
+            var lastToken = GetLastTokenInSelection();
+
+            var span = TextSpan.FromBounds(firstToken.SpanStart, lastToken.Span.End);
+
+            foreach (var node in SemanticDocument.Root.DescendantNodesAndSelf())
+            {
+                if (!node.Span.OverlapsWith(span))
+                {
+                    continue;
+                }
+
+                if (IsConfigureAwaitFalse(node) && !UnderAnonymousOrLocalMethod(node.GetFirstToken(), firstToken, lastToken))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private bool IsConfigureAwaitFalse(SyntaxNode node)
+        {
+            var syntaxFacts = SemanticDocument.Project.LanguageServices.GetService<ISyntaxFactsService>();
+            if (!syntaxFacts.IsInvocationExpression(node))
+            {
+                return false;
+            }
+
+            var invokedExpression = syntaxFacts.GetExpressionOfInvocationExpression(node);
+            if (!syntaxFacts.IsSimpleMemberAccessExpression(invokedExpression))
+            {
+                return false;
+            }
+
+            var name = syntaxFacts.GetNameOfMemberAccessExpression(invokedExpression);
+            var identifier = syntaxFacts.GetIdentifierOfSimpleName(name);
+            if (!syntaxFacts.StringComparer.Equals(identifier.ValueText, nameof(Task.ConfigureAwait)))
+            {
+                return false;
+            }
+
+            var arguments = syntaxFacts.GetArgumentsOfInvocationExpression(node);
+            if (arguments.Count != 1)
+            {
+                return false;
+            }
+
+            var argument = arguments[0];
+            var expression = syntaxFacts.GetExpressionOfArgument(argument);
+            return syntaxFacts.IsFalseLiteralExpression(expression);
```

Comment:
💡 Consider adding a test for a named argument:

```
await x.ConfigureAwait(continueOnCapturedContext: false)
```

I have seen this pattern in real code. It should already be handled by `GetExpressionOfArgument` but we should make sure.